rholang/src/main/scala/coop/rchain/rholang/interpreter/storage/ChargingRSpace.scala

```diff
@@ -0,0 +1,88 @@
+package coop.rchain.rholang.interpreter.storage
+import cats.implicits._
+import coop.rchain.models.{BindPattern, Channel, ListChannelWithRandom, TaggedContinuation}
+import coop.rchain.rholang.interpreter.Runtime.RhoPureSpace
+import coop.rchain.rholang.interpreter.accounting.CostAccount._
+import coop.rchain.rholang.interpreter.accounting.{CostAccount, CostAccountingAlg}
+import coop.rchain.rholang.interpreter.errors
+import coop.rchain.rholang.interpreter.errors.OutOfPhlogistonsError
+import coop.rchain.rspace.pure.PureRSpace
+import coop.rchain.rspace.{Blake2b256Hash, Checkpoint, Match}
+import monix.eval.Task
+
+import scala.collection.immutable.Seq
+
+object ChargingRSpace {
+  def pureRSpace(implicit costAlg: CostAccountingAlg[Task], space: RhoPureSpace) =
+    new PureRSpace[Task,
+                   Channel,
+                   BindPattern,
+                   OutOfPhlogistonsError.type,
+                   ListChannelWithRandom,
+                   ListChannelWithRandom,
+                   TaggedContinuation] {
```

Comment:
This object is unused and seems as if it needs some more love (use type aliases, remove commented out code, maybe elliminate some TODOs). It also lacks tests. Since it's not used yet, please remove this commit and propose the change with a PR that will use it.